### DISCUSSION TITLE
Custom Filtering for Large Data

### DISCUSSION THREAD

**Discussion by louisgurung** [Megabit/Blazorise#2436] [Q&A]
So I called my service, sent search parameter, sort parameter everything on OnReadData as advised for Large Data. The problem is for filtering, I cannot use OnCustomFilter as I want to search all the pages. When I create a TextEdit and bind customFilterValue to it, typing something doesn't trigger datagrid reload but only changing page will give me the correct result.

**Comment by stsrki** [comment]
You can manually trigger DataGrid reload.

eg.

```cshtml
<DataGrid @ref="@dataGridRef" ...>
@code{
  DataGrid dataGridRef;

  Task OnSomeMethod()
  {
    return dataGridRef.Reload();
  }
}
```

**Reply by stsrki** [reply]
Well, large data was never supposed to work with custom filtering. That is the first problem. Have you tried just updating `Data` and `TotalItems` directly when search changes?

```cshtml
<TextEdit Text="@customFilterValue" TextChanged="@OnCustomFilterValue" />
@code{
  async OnCustomFilterValue(string value)
  {
    var result = await someApi.GetData();
    FilteredList = result.Data;
    totalDataCount = result.Total;
  }
}
```

**Reply by louisgurung** [reply]
Thankyou @stsrki. I deleted my comment earlier as it is working now. I haven't used custom filtering. I used a TextField for search, binded it to @customfiltervalue and reloaded grid @onChange like you said. I have handled all logic(Sorting, Searching, Pagination) in @OnReadData. It worked for me.
Thanks a lot.
